shared/constants/types/flow-types.js

```diff
@@ -5831,7 +5831,8 @@ export type metadataGetMetadataRpcParam = Exact<{
   startRevision: long,
   stopRevision: long,
   logTags: {[key: string]: string},
-  lockBeforeGet?: ?LockID
+  lockBeforeGet?: ?LockID,
+  noCreate: boolean
```

Comment:
can you have a required parameter after an optional parameter?

Reply:
I think so; putMetadata already does that. This is a struct rather than function parameter list so shouldn't matter right?

Reply:
I think it does matter, since structs are encoded over RPC in array format, which depends on order. But if `putMetadata` does that successfully, then it must work :)

Reply:
Hmm are you sure it's in array format? I thought they were msgpack dictionaries with named fields so the order wouldn't matter.

Reply:
I think you're right: we encode structs as dictionaries, but when we actually put them on the wire, the RPC params are encoded in array format. But that's one level up from what we're talking about here.

Reply:
> the RPC params are encoded in array format

Hmm I was under the impression RPC function parameters were all grouped into an argument struct. This "struct" is actually the parameter list itself.

Reply:
Yes, but the RPC has some metadata fields above that struct itself. Those fields are encoded in array format.

Reply:
Oh I see; that makes sense. Thanks.